### PR TITLE
Multi-GPU fixes

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## 📝 Description

Testing script

```bash
#!/bin/bash

models=("Cfa" "Cflow" "Csflow" "Dfkde" "Dfm" "Draem" "Dsr" "EfficientAd" "Fastflow" "Fre" "Ganomaly" "Padim" "Patchcore" "ReverseDistillation" "Rkde" "Stfpm" "Uflow" "VlmAd" "WinClip" "AiVad")

# Loop through each model and run the anomalib train command
for model in "${models[@]}"; do
    anomalib train --model "$model" --data MVTec --trainer.max_epochs 2 --trainer.devices 2 --trainer.strategy='ddp_find_unused_parameters_true'
done
```

## Works

1. CFA
2. CFlow
3. CSFlow
4. Dfkde
5. Dfm
6. Dsr
7. Fastflow
8. Ganomaly
9. Padim
10. Patchcore
11. ReverseDistillation
12. Stfpm
13. Uflow
14. WinCLIP
15. EfficientAd
16. VlmAd
17. AiVad  // visualization stage does not work 
18. Draem
19. Fre

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/data/validators/torch/video.py [hunk 1]
```diff
@@ -600,17 +600,18 @@ def validate_gt_mask(mask: torch.Tensor | None) -> Mask | None:
         if mask is None:
```
Comment (ashwinvaidya17) [inline src/anomalib/data/validators/torch/video.py:600]: We need to revisit the docstrings for this method

**Review by samet-akcay — APPROVED** [review]
looking good, thanks 🔥

**Comment by haimat** [comment]
Awesome, these are great news, thanks a lot!
When do you plan to release an official update with this change?

**Comment by samet-akcay** [comment]
Our plan is to pre-release before the 20th

**Comment by haimat** [comment]
Hello, we are so much looking forward to multi-GPU training :)
Do you know when you can release this change?

**Comment by samet-akcay** [comment]
Hi, as soon as passing this CI :)
https://github.com/openvinotoolkit/anomalib/pull/2465

**Comment by haimat** [comment]
> Hi, as soon as passing this CI :) #2465

Awesome, thanks!

**Comment by blaz-r** [comment]
> Hi, as soon as passing this CI :) #2465
> 
> We didn't release it on the 20th of December mainly because we thought the documentation is not sufficient. We worked on the documentation during Christmas.
> 
> @blaz-r kindly added a new algorithm to v2 as well, but one of the tests are failing now. As soon as we fix the test, we'll release, which we hope to sort out by the end of today.

I think I found the issue, I'm going to open a PR in a few mins. Apologies about that.

**Comment by samet-akcay** [comment]
> > Hi, as soon as passing this CI :) #2465
> > We didn't release it on the 20th of December mainly because we thought the documentation is not sufficient. We worked on the documentation during Christmas.
> > @blaz-r kindly added a new algorithm to v2 as well, but one of the tests are failing now. As soon as we fix the test, we'll release, which we hope to sort out by the end of today.
> 
> I think I found the issue, I'm going to open a PR in a few mins. Apologies about that.

no worries @blaz-r, I think I've fixed it already, let's wait for the test results

**Comment by blaz-r** [comment]
Good @samet-akcay, that might fix it, but in case it doesn't I also opened a PR that also adds `device=` to that same lines in #2490 .

**Comment by samet-akcay** [comment]
> Good @samet-akcay, that might fix it, but in case it doesn't I also opened a PR that also adds `device=` to that same lines in #2490 .

Looks like it is passing 
https://github.com/openvinotoolkit/anomalib/actions/runs/12670086114/job/35308989508?pr=2465

**Comment by blaz-r** [comment]
Great! 😄

**Comment by samet-akcay** [comment]
You could try it with `pip install anomalib==2.0.0b2`

**Comment by haimat** [comment]
Thanks, that is great news - I will give it a try :+1: 
Do you have any docs on the breaking API changes?

**Comment by haimat** [comment]
@samet-akcay I tried to get beta 2 up and running, however, I am not sure how to update our workflow.
For example, neither the `Folder` nor the `Engine` classes accept the `task` argument any more.
How else do I specify the task?
Could you provide a short example of how a basic classificiation training with a given folder would look like in version 2?

**Comment by haimat** [comment]
@samet-akcay Heyho, any news on this?

**Comment by djdameln** [comment]
> @samet-akcay I tried to get beta 2 up and running, however, I am not sure how to update our workflow. For example, neither the `Folder` nor the `Engine` classes accept the `task` argument any more. How else do I specify the task? Could you provide a short example of how a basic classificiation training with a given folder would look like in version 2?

Hi, you can just omit the `task` parameter when constructing the dataset. Anomalib will automatically recognize that no ground truth masks are available, and skip the pixel-level metric computation in the evaluation sequence. 

During inference, the model will produce both pixel-level and image-level predictions. If you want to perform classification, you can just ignore the pixel-level predictions.

**Comment by haimat** [comment]
@djdameln Thanks for your response and info on `task`.
Do you have a notebook or some kind of minimal example of how to train a model in new anomalib v2.0?

**Pull request merged**